hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java

```diff
@@ -70,32 +80,21 @@ public void migrate(@NonNull final MigrationContext ctx) {
         requireNonNull(ctx);
         final WritableKVState<EntityNumber, Node> writableNodes =
                 ctx.newStates().get(NODES_KEY);
-        ReadableKVState<AccountID, Account> readableAccounts = null;
-        try {
-            readableAccounts = ctx.newStates().get(ACCOUNTS_KEY);
-        } catch (IllegalArgumentException e) {
-            log.info("AccountStore is not found, can be ignored.");
-        }
+
         final var networkInfo = ctx.networkInfo();
         final var addressBook = networkInfo.addressBook();
         final var bootstrapConfig = ctx.configuration().getConfigData(BootstrapConfig.class);
-        final var accountConfig = ctx.configuration().getConfigData(AccountsConfig.class);
-        var adminKey = Key.DEFAULT;
-        if (readableAccounts != null) {
-            final var adminAccount = readableAccounts.get(AccountID.newBuilder()
-                    .accountNum(accountConfig.addressBookAdmin())
-                    .build());
-            if (adminAccount != null) {
-                adminKey = adminAccount.keyOrElse(Key.DEFAULT);
-            }
-        }
+
         log.info("Started migrating nodes from address book");
+        final var adminKey = getAccountAdminKey(ctx);
+        final var nodeDetailMap = getNodeAddressMap(ctx);
 
         Key finalAdminKey = adminKey == null || adminKey.equals(Key.DEFAULT)
                 ? Key.newBuilder().ed25519(bootstrapConfig.genesisPublicKey()).build()
                 : adminKey;
-        addressBook.forEach(nodeInfo -> {
-            final var node = Node.newBuilder()
+        NodeAddress nodeDetail;
```

Comment:
Why would nodeDetails have the information ? On first migration you send Synthetic update to nodeDetails while handling transaction right ?